src/Core/Domain/Product/Stock/Query/AbstractGetStockMovementHistory.php

```diff
@@ -28,17 +28,13 @@
 
 namespace PrestaShop\PrestaShop\Core\Domain\Product\Stock\Query;
 
-use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use LogicException;
+use PrestaShop\PrestaShop\Core\Domain\Shop\Exception\ShopException;
 use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
 
-class GetEmployeesStockMovements
+abstract class AbstractGetStockMovementHistory
 {
-    public const DEFAULT_LIMIT = 5;
-
-    /**
-     * @var ProductId
-     */
-    private $productId;
+    protected const DEFAULT_LIMIT = 5;
```

Comment:
That's a technical default value, not a configuration. A JS component should rely on the use case value.
I would remove totally and provide the size explicitely, rather than hard coding a behaviour.

Reply:
Yep I agree with you, but we often need a default value even for JS component For example when you propose a sector to choose the size of the pagination It offers a restricted list of possible limits, and a default one is selected by default.

In such cases we often use the default limit from the PHP code so that FO and BO are synced on this value, it's convenient because you can change the default value in one place. But in this case anyway we don't have the possibility to select the number of elements so it's not that important.